src/main/java/meeju/TaskList.java

```diff
@@ -0,0 +1,241 @@
+package meeju;
+
+import java.util.ArrayList;
+
+/**
+ * TaskList class manages a list of tasks using an ArrayList and provides methods to manipulate this list.
+ * It supports adding, marking, unmarking, finding and deleting tasks.
+ * It also interacts with the Storage class to permenantly store the tasks in disk upon every update to the list.
+ */
+public class TaskList {
+    private ArrayList<Task> taskList;
+
+    private Storage storage;
+
+    /**
+     * Construct for TaskList object.
+     * Creates a new Storage object and initialises the ArrayList with the tasks saved in the disk.
+     *
+     * @param storage The Storage object responsible for loading and saving tasks.
+     */
+    public TaskList(Storage storage) {
+        this.storage = storage;
+        this.taskList = this.storage.initialiseList();
+    }
+
+    public Task getTask(int index) {
+        return this.taskList.get(index);
+    }
+
+
+    public int getNumberOfTask() {
```

Comment:
Should the method name be changed to make it clearer (e.g. `getNumberOfTasks()` )?